**Anjala Emmadishetti**
3/05/2024

In [27]:
import pandas as pd
from sklearn.model_selection import train_test_split

# Load the data
data = pd.read_excel("/content/college_admissions_2022_v3.xlsx")

# Report the size of the dataset
print("Size of the dataset before dropping missing values:")
print(data.info())

# Drop missing values
data.dropna(inplace=True)

# Report the size of the remaining dataset
print("\nSize of the dataset after dropping missing values:")
print(data.info())

# Drop specified variables
data.drop(['UNITID', 'INSTNM', 'Appl', 'Enroll_fall', 'Enroll_ratio', 'Enroll_tot'], axis=1, inplace=True)

# Split the data into training and test sets
X = data.drop('Low_enroll_Ratio', axis=1)
y = data['Low_enroll_Ratio']

X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.4, random_state=101)

# Print the shapes of the training and test sets
print("\nShape of X_train:", X_train.shape)
print("Shape of X_test:", X_test.shape)
print("Shape of y_train:", y_train.shape)
print("Shape of y_test:", y_test.shape)

Size of the dataset before dropping missing values:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1733 entries, 0 to 1732
Data columns (total 18 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   UNITID            1733 non-null   int64  
 1   INSTNM            1733 non-null   object 
 2   Public            1733 non-null   int64  
 3   Appl              1733 non-null   int64  
 4   Enroll_fall       1733 non-null   int64  
 5   Enroll_ratio      1733 non-null   float64
 6   Low_enroll_Ratio  1733 non-null   int64  
 7   Enroll_tot        1733 non-null   int64  
 8   GRTOTLT           1723 non-null   float64
 9   Tuition_in        1733 non-null   int64  
 10  Tuition_out       1733 non-null   int64  
 11  Stufacr           1733 non-null   int64  
 12  RRFTCT            1733 non-null   int64  
 13  Salary            1712 non-null   float64
 14  Ug_enter          1733 non-null   int64  
 15  Sat               1002 non-null   flo

Memory usage before dropping missing values is <b> 243.8+ KB</b>, Memory usage after droping missing values is <b>140.9+ KB</b>
<br>

This reduction in memory usage suggests that dropping missing values has helped optimize memory utilization by eliminating unnecessary storage for those missing values.


Decision Tree Classifier

In [25]:

data.head()

Public  Low_enroll_Ratio  GRTOTLT  Tuition_in  Tuition_out  Stufacr  \
7        0                 1      1.0       57120        57120       10   
9        0                 1   1515.0       61989        61989        6   
10       0                 1   1659.0       52659        52659        7   
12       0                 1   2670.0       59100        59100       16   
14       0                 1    414.0       58928        58928        8   

    RRFTCT    Salary  Ug_enter    Sat     Room   Books  
7      593  110874.0       422  790.0  17016.0  1250.0  
9     1592  246477.0      2233  800.0  16286.0  1366.0  
10    1949  264272.0      2719  800.0  19502.0   900.0  
12    4492  182755.0      3234  790.0  18430.0  1000.0  
14     454  162931.0       436  790.0  18026.0   760.0

In [28]:
from sklearn.tree import DecisionTreeClassifier
dtree = DecisionTreeClassifier(max_depth=3, random_state=1)
dtree.fit(X_train, y_train)

DecisionTreeClassifier(max_depth=3, random_state=1)

Random Forest Classifier

In [29]:
from sklearn.ensemble import RandomForestClassifier

rfc = RandomForestClassifier(n_estimators=500, max_depth=3, random_state=1)
rfc.fit(X_train, y_train)

RandomForestClassifier(max_depth=3, n_estimators=500, random_state=1)

Prediction and Evaluation

In [30]:
dtree_predictions = dtree.predict(X_test)
rfc_predictions = rfc.predict(X_test)

 Classification report

In [31]:
  print("Decision Tree Classifier:")
print(classification_report(y_test, dtree_predictions))

print("\nRandom Forest Classifier:")
print(classification_report(y_test, rfc_predictions))

Decision Tree Classifier:
              precision    recall  f1-score   support

           0       0.25      0.17      0.20         6
           1       0.99      0.99      0.99       374

    accuracy                           0.98       380
   macro avg       0.62      0.58      0.59       380
weighted avg       0.98      0.98      0.98       380


Random Forest Classifier:
              precision    recall  f1-score   support

           0       0.50      0.17      0.25         6
           1       0.99      1.00      0.99       374

    accuracy                           0.98       380
   macro avg       0.74      0.58      0.62       380
weighted avg       0.98      0.98      0.98       380



Both the mode­ls work very well, with an accuracy rate of 98% for both.

The­ Decision Tree Classifie­r does not work as well for the class 0, with lower scores for precision 25%, re­call ,17%, and F1-score 20%. However, the­ Random Forest Classifier performs be­tter for this category, with precision at 50%, re­call at 17%, and F1-score at 25%.
<br>

But for class 1, both models do an exceptional job, with pre­cision, recall, and F1-score around 99%.

When looking at the­ overall performance score­s like macro average and we­ighted average, the­ Random Forest Classifier does slightly be­tter than the Decision Tre­e Classifier.